test/smoke/src/areas/terminal/terminal-tabs.test.ts

```diff
@@ -41,99 +29,100 @@ export function setup(opts: ParsedArgs) {
+			await terminal.runCommand(TerminalCommandId.CreateNew);
+			await terminal.clickPlusButton();
+			await terminal.assertTerminalGroups([[{}], [{}]]);
 		});
 
 		it('should update color of the single tab', async () => {
 			await terminal.runCommand(TerminalCommandId.Show);
 			const color = 'Cyan';
-			await terminal.runCommand(TerminalCommandId.ChangeColor, color);
-			const singleTab = await code.waitForElement('.single-terminal-tab');
-			ok(singleTab.className.includes(`terminal-icon-terminal_ansi${color}`));
+			await terminal.runCommandWithValue(TerminalCommandIdWithValue.ChangeColor, color);
+			await terminal.assertSingleTab({ color });
 		});
 
 		it('should update color of the tab in the tabs list', async () => {
 			await terminal.runCommand(TerminalCommandId.Show);
 			await terminal.runCommand(TerminalCommandId.Split);
-			const tabs = await terminal.getTabLabels(2);
-			ok(tabs[0].startsWith('┌'));
-			ok(tabs[1].startsWith('└'));
 			const color = 'Cyan';
-			await terminal.runCommand(TerminalCommandId.ChangeColor, color);
-			await code.waitForElement(`.terminal-tabs-entry .terminal-icon-terminal_ansi${color}`);
+			await terminal.runCommandWithValue(TerminalCommandIdWithValue.ChangeColor, color);
+			await terminal.assertTerminalGroups([[{}, { color }]]);
 		});
 
 		it('should update icon of the single tab', async () => {
 			await terminal.runCommand(TerminalCommandId.Show);
 			const icon = 'symbol-method';
-			await terminal.runCommand(TerminalCommandId.ChangeIcon, icon);
-			await code.waitForElement(`.single-terminal-tab .codicon-${icon}`);
+			await terminal.runCommandWithValue(TerminalCommandIdWithValue.ChangeIcon, icon);
+			await terminal.assertSingleTab({ icon });
 		});
 
 		it('should update icon of the tab in the tabs list', async () => {
 			await terminal.runCommand(TerminalCommandId.Show);
 			await terminal.runCommand(TerminalCommandId.Split);
-			const tabs = await terminal.getTabLabels(2);
-			ok(tabs[0].startsWith('┌'));
-			ok(tabs[1].startsWith('└'));
 			const icon = 'symbol-method';
-			await terminal.runCommand(TerminalCommandId.ChangeIcon, icon);
-			await code.waitForElement(`.terminal-tabs-entry .codicon-${icon}`);
+			await terminal.runCommandWithValue(TerminalCommandIdWithValue.ChangeIcon, icon);
+			await terminal.assertTerminalGroups([[{}, { icon }]]);
 		});
 
 		it('should rename the single tab', async () => {
 			await terminal.runCommand(TerminalCommandId.Show);
 			const name = 'my terminal name';
-			await terminal.runCommand(TerminalCommandId.Rename, name);
-			await code.waitForElement('.single-terminal-tab', e => e ? e?.textContent.includes(name) : false);
+			await terminal.runCommandWithValue(TerminalCommandIdWithValue.Rename, name);
+			await terminal.assertSingleTab({ name });
 		});
 
 		it('should rename the tab in the tabs list', async () => {
 			await terminal.runCommand(TerminalCommandId.Show);
 			await terminal.runCommand(TerminalCommandId.Split);
+			await timeout(500);
```

Comment:
Timeouts are inherently flaky, can this be awaiting for a certain number of tabs to show up?